### PR TITLE
addUser

### PR DESCRIPTION
**Pull request by cayennegoh**
add the ability to add users who are the people who borrowed books from the library and list the list of users.

### CODE REVIEW

File: src/main/java/bookmarked/command/AddUserCommand.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        this.splitItem = splitItem;
+        this.bookDataFile = bookDataFile;
+    }
+
+    @Override
+    public void handleCommand() {
+        assert newItem != null : "Item should not be null";
+        String[] newSplitUser = this.newItem.split("adduser");
+        try {
+            processAddUserCommand(newSplitUser, listOfUsers);
```
Comment (Elyovs) [inline src/main/java/bookmarked/command/AddUserCommand.java:31]: good that you factor out into another method

File: src/main/java/bookmarked/command/ListUserCommand.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public void handleCommand() {
+        this.splitCommand = inputCommand.split("listuser ");
+
+        //This will be updated to switch/case when the date feature is added
+        try {
+            if (inputCommand.matches("listuser")) {
+                runListCommand();
+            }
+        } catch (EmptyListException e) {
+            Ui.printEmptyListMessage();
```
Comment (Elyovs) [inline src/main/java/bookmarked/command/ListUserCommand.java:33]: A lot of the exception are handled properly. good job!

File: src/main/java/bookmarked/parser/Parser.java [hunk 3]
```diff
@@ -1,7 +1,8 @@
 package bookmarked.parser;
 
 import bookmarked.Book;
-import bookmarked.command.EditCommand;
```
Comment (eliztan) [inline src/main/java/bookmarked/parser/Parser.java:4]: Why is the editcommand remmoved from import when it's called later on in parser?

File: src/main/java/bookmarked/parser/Parser.java [hunk 4]
```diff
@@ -12,25 +13,26 @@
 import bookmarked.command.BorrowCommand;
 import bookmarked.command.HelpCommand;
 import bookmarked.command.ListCommand;
+import bookmarked.command.ListUserCommand;
 import bookmarked.exceptions.BookMarkedException;
-import bookmarked.command.ExtendCommand ;
```
Comment (eliztan) [inline src/main/java/bookmarked/parser/Parser.java:16]: similar to before, add back ExtendCommand import

File: src/main/java/bookmarked/command/BorrowCommand.java [hunk 5]
```diff
@@ -33,14 +38,32 @@ public class BorrowCommand extends Command {
      * @param listOfBooks  The list of books to be searched for borrowing.
      * @param bookDataFile The file where book data is stored.
      */
-    public BorrowCommand(String[] commandParts, ArrayList<Book> listOfBooks, File bookDataFile) {
+    public BorrowCommand(String[] commandParts, ArrayList<Book> listOfBooks, File bookDataFile,
+                         ArrayList<User> listOfUsers,String newItem) throws EmptyArgumentsException{
```
Comment (Elyovs) [inline src/main/java/bookmarked/command/BorrowCommand.java:42]: minor thing, add space between , and String

File: src/main/java/bookmarked/command/ListUserCommand.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        for (User user : listOfUsers) {
+            System.out.println("User: " + user.getName());
+            System.out.print("Borrowed Books: ");
+            if (user.getUserBooks().isEmpty()) {
+                System.out.println("None");
+            } else {
+                for (int i = 0; i < user.getUserBooks().size(); i++) {
+                    System.out.print(user.getUserBooks().get(i).getName());
+                    if (i < user.getUserBooks().size() - 1) {
+                        System.out.print(", ");
```
Comment (Elyovs) [inline src/main/java/bookmarked/command/ListUserCommand.java:40]: next time, can make extract method on this (deep nesting)

**Pull request merged**